README.md

```diff
@@ -330,10 +369,85 @@ css file that styles the "pygmentized" code examples. We use
 blocks
 [here](https://github.com/trulia/hologram-example/tree/gh-pages/hologram_assets/doc_assets/css).
 
+### Custom Code Example Renderers
+
+By default, hologram supports the following code example types:
+
+- `html_example` and `html_example_table`
+- `haml_example` and `haml_example_table`
+- `slim_example` and `slim_example_table`
+- `js_example`
+- `jsx_example`
+
+Let's say you want to include coffeescript examples in your styleguide.
+You'll need to create a custom renderer for this.
+
+You can generate custom examples with the code example renderer factory:
+
+```ruby
+require 'coffee-script'
+
+Hologram::CodeExampleRenderer::Factory.define('coffee') do
+  example_template 'js_example_template'
+  table_template 'js_table_template'
```

Comment:
@stubbornella @bebepeng yeah, we should mention something about creating custom template files. Maybe we should even name it something like `my_custom_coffee_example_template` and `my_custom_coffee_table_template` just to make it less confusing.